pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -686,6 +694,34 @@ func checkForPodSecondaryNetworkAttachment(pod *corev1.Pod) (string, bool) {
 	return netObj, netObjExists && netObj != ""
 }
 
+// When a Kubernetes Node reboots and the OVSDB file is not properly restored,
+// the primary and secondary OVS ports in OVSDB are lost. As a result, Pods may
+// temporarily appear as "unknown" and later get recreated with new container IDs.
+// The cleanup of secondary interfaces during agent restart depends on the InterfaceStores
+// being initialized with existing OVS ports. If the OVSDB file is missing, the
+// primary InterfaceStore is empty, preventing the secondary InterfaceStore from
+// initializing correctly. Consequently, secondary IPs assigned to old Pods (with
+// previous container IDs) are not deleted. To resolve this, all IPPool allocations
+// must be checked, and any IPs associated with non-existing container IDs released.
+// Otherwise, stale secondary IPs remain in the IPPool and cannot be reused.
+func (pc *PodController) cleanUpStaleIPAddresses() {
+	pools, _ := pc.ipPoolLister.List(labels.Everything())
+	for _, ipPool := range pools {
+		for _, address := range ipPool.Status.IPAddresses {
+			if _, err := pc.podLister.Pods(address.Owner.Pod.Namespace).Get(address.Owner.Pod.Name); err == nil {
```

Comment:
is there a guarantee that `address.Owner.Pod` is non nil here? Could it be owned by a StatefulSet instead?